lib/app/backend/BackendProcess.js

```diff
@@ -1,26 +1,62 @@
 const io = require('socket.io-client')
+const async = require('neo-async')
 const logger = require('../../logger')
+const AppSettings = require('../AppSettings')
+const UserSettings = require('../../user/UserSettings')
+const ExtensionWatcher = require('../../ExtensionWatcher')
 
 class BackendProcess {
   constructor () {
     this.dcAddress = process.env.SGCLOUD_DC_WS_ADDRESS || 'https://developer-connector.shopgate.cloud'
     this.socket = null
+    this.extensionWatcher = new ExtensionWatcher()
   }
 
   connect (cb) {
     logger.info('Establishing SDK connection')
-    this.socket = io(this.dcAddress)
+    this.socket = io(this.dcAddress, { extraHeaders: {
+      Authorization: 'Bearer ' + UserSettings.getInstance().getSession().token
+    }})
 
     this.socket.on('connect', () => {
```

Comment:
please cleanup this whole block, its a mess of events and function calls and really not easy to read